### ISSUE TITLE
Grave accents causing visual problem with multiline selection

### ISSUE THREAD

**Issue by rm-code** [atom/atom#14622]
### Prerequisites

Reproduced it in `--safe` mode. (Didn't have time for the other steps).

### Description

Cursors are misplaced after selecting multiple lines for the first time and typing in grave accents (`). As far as I can tell it is just a visual bug as the next characters are placed correctly behind the accents.

![foo](https://cloud.githubusercontent.com/assets/11627131/26406077/87e5e334-4097-11e7-9350-b284fac8a459.gif)

As you can see in the gif, the cursors are placed correctly after the initial faulty placement.

### Steps to Reproduce

See the gif above. Reproducible each time.

*Edited by @rsese, updated steps:*

1. Open new Window
2. Enter the following lines
```
foo
bar
baz
```
(Don't move the cursor after you are done. It should be right behind baz)

3. Selection -> Add selection above
4. Selection -> Add selection above

You should now have all three lines selected (IMPORTANT: Don't move the cursor. The bug only happens if you add the grave accents right after the "Add selection above" action)
<img width="272" alt="bildschirmfoto 2017-10-27 um 23 03 06" src="https://user-images.githubusercontent.com/11627131/32125205-17f0d0b2-bb6b-11e7-926f-3a483dbc1aab.png">

5. Add the grave (`) character (on macOS, type <kbd>option-e</kbd>

The result should look like this:
<img width="271" alt="bildschirmfoto 2017-10-27 um 23 06 03" src="https://user-images.githubusercontent.com/11627131/32125277-7335563c-bb6b-11e7-9d77-7f0217e56899.png">

### Versions

```
Atom    : 1.17.0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
````

```
apm  1.17.0
npm  3.10.5
node 6.9.5 x64
python 2.7.10
git 2.12.1
```

```
macOS Sierra 10.12.4
```

**Comment by rsese** [comment]
Thanks for the report!  I just tried this in 1.17.1 on macOS 10.12.5 and I didn't see the same behavior - do you see the same thing if the file is set to plain text?

**Comment by rm-code** [comment]
> do you see the same thing if the file is set to plain text?

Yes, it happens in plain text mode too.

I'm installing macOS 10.12.5 atm and will upgrade to Atom 1.17.2 before testing again.

**Comment by rm-code** [comment]
@rsese It still happens for me. Which theme did you use for testing? I wonder if it is some kind of font related issue?

**Comment by rsese** [comment]
I'm using One Dark - I also double checked that I didn't change any of the font settings in Settings.

**Comment by rm-code** [comment]
<img width="120" alt="bildschirmfoto 2017-10-18 um 02 37 57" src="https://user-images.githubusercontent.com/11627131/31695887-6788b9b6-b3ad-11e7-80c7-484918f1df38.png">

Just a small update to show that this is still happening with Atom 1.21.1 under MacOS 10.12.6.

**Comment by rm-code** [comment]
Also I just noticed this only happens if you enter the grave accents directly after applying the multi-line selection. If you move the cursor up and back down and then enter the accents the cursor is displayed correctly.

**Comment by rsese** [comment]
Thanks for the update @rm-code - can you share exact steps to reproduce so I can give it a try again?  As detailed as possible would be great just so I'm sure I'm trying things like you are.

**Comment by rm-code** [comment]
@rsese Try this:

1. Open new Window
2. Enter the following lines
```
foo
bar
baz
```
(Don't move the cursor after you are done. It should be right behind baz)

3. Selection -> Add selection above
4. Selection -> Add selection above

You should now have all three lines selected (IMPORTANT: Don't move the cursor. The bug only happens if you add the grave accents right after the "Add selection above" action)
<img width="272" alt="bildschirmfoto 2017-10-27 um 23 03 06" src="https://user-images.githubusercontent.com/11627131/32125205-17f0d0b2-bb6b-11e7-926f-3a483dbc1aab.png">

5. Add the grave (`) character

The result should look like this:
<img width="271" alt="bildschirmfoto 2017-10-27 um 23 06 03" src="https://user-images.githubusercontent.com/11627131/32125277-7335563c-bb6b-11e7-9d77-7f0217e56899.png">

**Comment by rsese** [comment]
Ahhh great, thanks for the detailed instructions I can reproduce now - looks like this happens with all lines except the first and last?

![select-cursors](https://user-images.githubusercontent.com/734194/32209823-4bc0786a-bdd8-11e7-9900-93a08cc8314d.gif)

>  If you move the cursor up and back down and then enter the accents the cursor is displayed correctly.

Also noticed if I just switch tabs or switch to another application and then back to Atom, the cursors are in the right place after that.

![select-cursors-reset](https://user-images.githubusercontent.com/734194/32209854-69c12300-bdd8-11e7-8a8a-f48f4d907f9d.gif)

I updated your steps to reproduce, thanks again :v:

**Comment by rm-code** [comment]
@as-cii 

There still seems to be an issue with the multi-selection. Should I open a new ticket for it?
![bug_2](https://user-images.githubusercontent.com/11627131/34820344-8121fd50-f6c0-11e7-8273-1ae9b30b9eec.gif)

**Comment by Ben3eeE** [comment]
@rm-code The fix for this issue is currently scheduled to be released in Atom 1.25. Do you see this issue when running Atom on the latest master branch?

**Comment by rm-code** [comment]
@Ben3eeE My apologies, I thought it was part of the 1.23.3 update :/
